modules/shared/src/main/scala/org/tessellation/merkletree/MerkleTree.scala

```diff
@@ -0,0 +1,128 @@
+package org.tessellation.merkletree
+
+import cats.data.NonEmptyList
+import cats.kernel.Eq
+import cats.syntax.eq._
+import cats.syntax.option._
+
+import org.tessellation.security.hash.Hash
+
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegInt
+
+case class ProofEntry(target: Hash, siblings: Either[Hash, Hash])
```

Comment:
Regarding `ProofEntry` class maybe we should have the second parameter called `sibling` instead of `siblings` here given that there is only one value there (either left or right sibling).

Reply:
:+1: